src/dtos/unit.dto.ts

```diff
@@ -93,19 +100,110 @@ export class UnitDTO {
   endDate: Date;
 
   @ApiProperty({
-    description:
-      propertyMetadata.monitorLocationDTONonLoadBasedIndicator.description,
-    example: propertyMetadata.monitorLocationDTONonLoadBasedIndicator.example,
-    name:
-      propertyMetadata.monitorLocationDTONonLoadBasedIndicator.fieldLabels
-        .value,
+    description: propertyMetadata.unitDTOSourceCategoryCd.description,
+    example: propertyMetadata.unitDTOSourceCategoryCd.example,
+    name: propertyMetadata.unitDTOSourceCategoryCd.fieldLabels.value,
   })
   @IsOptional()
-  @IsInt()
-  @IsInRange(0, 1, {
-    message: (args: ValidationArguments) => {
-      return `The value of [${args.value}] for [${args.property}] must be an integer of 0 and 1 for [${KEY}].`;
-    },
+  @IsString()
+  sourceCategoryCd?: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOCommOpDate.description,
+    example: propertyMetadata.unitDTOCommOpDate.example,
+    name: propertyMetadata.unitDTOCommOpDate.fieldLabels.value,
   })
-  nonLoadBasedIndicator: number;
+  @IsOptional()
+  @IsDateString()
+  commOpDate?: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOComrOpDate.description,
+    example: propertyMetadata.unitDTOComrOpDate.example,
+    name: propertyMetadata.unitDTOComrOpDate.fieldLabels.value,
+  })
+  @IsOptional()
+  @IsDateString()
+  comrOpDate?: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOUserId.description,
+    example: propertyMetadata.unitDTOUserId.example,
+    name: propertyMetadata.unitDTOUserId.fieldLabels.value,
+  })
+  @IsNotEmpty()
+  @IsString()
+  userId: string;
+
+  @ApiProperty({
+    description: propertyMetadata.unitDTOAddDate.description,
+    example: propertyMetadata.unitDTOAddDate.example,
+    name: propertyMetadata.unitDTOAddDate.fieldLabels.value,
+  })
+  @IsNotEmpty()
+  @IsDateString()
+  addDate: string;
```

Comment:
Why is this date a string?

Reply:
This is converted from a `Date` to an ISO string in the `one` method of `UnitMap`. This is done for all `addDate` properties throughout the API.